src/utils/gen_util.rs

```diff
@@ -89,21 +80,63 @@ pub fn get_simple_byte_values(bytes: u64, spacing: bool) -> (f64, String) {
     }
 }
 
+/// Returns a tuple containing the value and the unit.  In units of 1024.
+/// This only supports up to a tebi.  Note the "single" unit will have a space appended to match the others if
+/// `spacing` is true.
+pub fn get_binary_prefix(quantity: u64, spacing: bool, unit: &str) -> (f64, String) {
+    match quantity {
+        b if b < KIBI_LIMIT => (
+            quantity as f64,
+            if spacing {
+                format!(" {}", unit)
+            } else {
+                unit.to_string()
+            },
+        ),
+        b if b < MEBI_LIMIT => (quantity as f64 / 1024.0, format!("Ki{}", unit)),
+        b if b < GIBI_LIMIT => (quantity as f64 / 1_048_576.0, format!("Mi{}", unit)),
+        b if b < TERA_LIMIT => (quantity as f64 / 1_073_741_824.0, format!("Gi{}", unit)),
+        _ => (quantity as f64 / 1_099_511_627_776.0, format!("Ti{}", unit)),
+    }
+}
+
+/// Returns a tuple containing the value and the unit.  In units of 1000.
+/// This only supports up to a tera.  Note the "single" unit will have a space appended to match the others if
+/// `spacing` is true.
+pub fn get_decimal_prefix(quantity: u64, spacing: bool, unit: &str) -> (f64, String) {
+    match quantity {
+        b if b < KILO_LIMIT => (
+            quantity as f64,
+            if spacing {
+                format!(" {}", unit)
+            } else {
+                unit.to_string()
+            },
+        ),
+        b if b < MEGA_LIMIT => (quantity as f64 / 1000.0, format!("K{}", unit)),
+        b if b < GIGA_LIMIT => (quantity as f64 / 1_000_000.0, format!("M{}", unit)),
+        b if b < TERA_LIMIT => (quantity as f64 / 1_000_000_000.0, format!("G{}", unit)),
+        _ => (quantity as f64 / 1_000_000_000_000.0, format!("T{}", unit)),
+    }
+}
+
 /// Gotta get partial ordering?  No problem, here's something to deal with it~
```

Comment:
Can probably replace this with OrderedFloat when we do a refactor in regards to the interpolation data structures.